### ISSUE TITLE
How to rerun strong baselines using the same video features on ActivityNet

### ISSUE THREAD

**Issue by sushizixin** [syuqings/video-paragraph#7]
Hi, syuqings, thank you for your repo. I am a little confused about how to run experiments of the strong baselines by using the same video features. For example, in VTransformer or MART, the features are sampled every 0.5 sec and the specific frame indices are loaded. When rerunning them, did you set the values of the sampling second for your provided features? Or did you use your framework and replace the method with the baselines? Could I know some details about this? Thank you very much.

**Comment by syuqings** [comment]
Hi, we just run the released code in the github of MART with a little modification.
In their code, a dictionary of self.frame_to_second is computed to record the real time of each feature (frm2sec), and the start and end frame index is computed by timestamp[0] \/ frm2sec and timestamp[1] \/ frm2sec.

We comment out the line of self.frame_to_second computing, and compute the start and end frame index for each clip by (timestamp[0] \/ duration) * total_frames, and (timestamp[1] \/ duration) * total_frames.

**Comment by sushizixin** [comment]
Got it, thanks for your reply!

**Comment by Kashu7100** [comment]
> 
> 
> Hi, we just run the released code in the github of MART with a little modification. In their code, a dictionary of self.frame_to_second is computed to record the real time of each feature (frm2sec), and the start and end frame index is computed by timestamp[0] / frm2sec and timestamp[1] / frm2sec.
> 
> We comment out the line of self.frame_to_second computing, and compute the start and end frame index for each clip by (timestamp[0] / duration) * total_frames, and (timestamp[1] / duration) * total_frames.

In this manner, the resulting features per event seems to be about 1/4 ~ 1/5 of the original features used in MART.
Ex. (36, 3072) vs (6, 4096)
MART sets 100 frames per event (not 100 frames per video) and thus the actual feature provided to the model becomes mostly masked. Did you repeat the features several times to match with the original or just use the 1/4 ~ 1/5 features to re-train the MART?

Thank you in advance.

**Comment by syuqings** [comment]
We didn't repeat the features and just use that to re-train the model.

**Comment by Kashu7100** [comment]
Hello, when I tried with your I3D feature, the scores actually improved compared to the original features in MART. However, METEOR, for example, was 15.90 (originally 15.52) and did not reach 17.12 as reported in the paper. Did you applied some other technique to improve the score or can you specify your procedure a little more?  

Thank you in advance.

**Comment by syuqings** [comment]
Hi, I have checked my training logs, and found that when I re-run the github of MART, the METEOR evaluation script didn't successfully return the score. So I commented out the line of METEOR computation and re-evaluated the METEOR with my own evaluator, and it reported the score of 17.12. I checked the difference of the two evaluators and guess that it may be due to the punctuation of ”."  Keeping the "." in the paragraph will make the METEOR score higher. I just found it out and I'm sorry to make the confusion.
